CI fails with test dependencies: AssertionError: Parameter has not changed

CI fails with test dependencies: https://github.com/huggingface/trl/actions/runs/20467967043/job/58816290636?pr=4745
> AssertionError: Parameter model.layers.0.self_attn.q_norm.weight has not changed
> AssertionError: Parameter model.layers.0.input_layernorm.weight has not changed

<details>

<summary>Failed tests</summary>

```python
  =========================== short test summary info ============================
  FAILED tests/test_dpo_trainer.py::TestDPOTrainer::test_train_with_iterable_dataset - AssertionError: assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train[trl-internal-testing/tiny-Qwen3MoeForSequenceClassification] - AssertionError: Parameter model.layers.0.self_attn.q_norm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train[trl-internal-testing/tiny-LlamaForSequenceClassification-3.2] - AssertionError: Parameter model.layers.0.input_layernorm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_dataset_types[standard_preference] - AssertionError: Parameter model.layers.0.input_layernorm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_dataset_types[conversational_preference] - AssertionError: Parameter model.layers.0.input_layernorm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_with_chat_template_kwargs - AssertionError: Parameter model.layers.0.input_layernorm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_reward_trainer.py::TestRewardTrainer::test_train_dataset_types[standard_implicit_prompt_preference] - AssertionError: Parameter model.layers.0.input_layernorm.weight has not changed
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_liger - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train[trl-internal-testing/tiny-Qwen3MoeForCausalLM] - AssertionError: Parameter model.layers.0.self_attn.v_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([[ 0.6758, -0.7031, -0.6289,  0.6953, -0.3594, -0.5586,  0.4082,  0.4316],\n        [-0.4199, -0.2490, -0.6406, -0.3047,  0.6406,  0.6367, -0.0221, -0.3379],\n        [ 0.4023,  0.0718,  0.2158, -0.3320,  0.0884,  0.4023,  0.2812,  0.5859],\n        [-0.1494, -0.4414, -0.3262,  0.6914, -0.3691, -0.6914,  0.5312, -0.5078]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[ 0.6758, -0.7031, -0.6289,  0.6953, -0.3594, -0.5586,  0.4082,  0.4316],\n        [-0.4199, -0.2490, -0.6406, -0.3047,  0.6406,  0.6367, -0.0221, -0.3379],\n        [ 0.4023,  0.0718,  0.2158, -0.3320,  0.0884,  0.4023,  0.2812,  0.5859],\n        [-0.1494, -0.4414, -0.3262,  0.6914, -0.3691, -0.6914,  0.5312, -0.5078]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_completion_only - AssertionError: Parameter model.layers.0.self_attn.q_norm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_non_chatml_conversational_data - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_model - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_pretokenized_data - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_assistant_only_and_completion_only - AssertionError: Parameter model.layers.0.self_attn.q_norm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_iterable_dataset - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_assistant_only_iterable_dataset - AssertionError: Parameter model.layers.0.self_attn.q_norm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_dft_loss - AssertionError: Parameter model.layers.0.input_layernorm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0', dtype=torch.bfloat16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0', dtype=torch.bfloat16,\n       requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_packing[bfd] - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_set_chat_template_from_model - AssertionError: Parameter gpt_neox.layers.1.post_attention_layernorm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0', dtype=torch.float16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0', dtype=torch.float16,\n       requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_moe_model_with_aux_loss - AssertionError: Parameter model.layers.0.self_attn.v_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([[ 0.6758, -0.7031, -0.6289,  0.6953, -0.3594, -0.5586,  0.4082,  0.4316],\n        [-0.4199, -0.2490, -0.6406, -0.3047,  0.6406,  0.6367, -0.0221, -0.3379],\n        [ 0.4023,  0.0718,  0.2158, -0.3320,  0.0884,  0.4023,  0.2812,  0.5859],\n        [-0.1494, -0.4414, -0.3262,  0.6914, -0.3691, -0.6914,  0.5312, -0.5078]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[ 0.6758, -0.7031, -0.6289,  0.6953, -0.3594, -0.5586,  0.4082,  0.4316],\n        [-0.4199, -0.2490, -0.6406, -0.3047,  0.6406,  0.6367, -0.0221, -0.3379],\n        [ 0.4023,  0.0718,  0.2158, -0.3320,  0.0884,  0.4023,  0.2812,  0.5859],\n        [-0.1494, -0.4414, -0.3262,  0.6914, -0.3691, -0.6914,  0.5312, -0.5078]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_set_chat_template_from_path - AssertionError: Parameter gpt_neox.layers.1.input_layernorm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0', dtype=torch.float16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0', dtype=torch.float16,\n       requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_packing[wrapped] - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_formatting_func - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f64a9bd6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f64a9bd6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm[trl-internal-testing/tiny-LlavaForConditionalGeneration] - AssertionError: Param model.vision_tower.vision_model.pre_layrnorm.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.float16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.float16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_toolcall_data - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_chat_template_kwargs - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm[trl-internal-testing/tiny-LlavaNextForConditionalGeneration] - AssertionError: Param model.image_newline is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([-0.3359,  0.0249, -0.2891, -0.0439,  0.0708,  0.2471,  0.1660,  0.0176,\n        -0.1875, -0.1348,  0.3105, -0.1465, -0.1299,  0.3770, -0.1040, -0.0447],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([-0.3359,  0.0249, -0.2891, -0.0439,  0.0708,  0.2471,  0.1660,  0.0176,\n        -0.1875, -0.1348,  0.3105, -0.1465, -0.1299,  0.3770, -0.1040, -0.0447],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_peft_with_quantization - RuntimeError: Float did not match BFloat16
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_with_gradient_checkpointing - AssertionError: Parameter model.layers.0.self_attn.q_proj.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([[-0.1196, -0.5664,  0.3203, -0.5898,  0.1914, -0.3730,  0.1631,  0.4121],\n        [-0.4941,  0.4980,  0.3828,  0.3633,  0.2207, -0.2539, -0.1533, -0.5547],\n        [ 0.1338, -0.2393,  0.4160,  0.1289,  0.0239, -0.1816,  0.2969, -0.3691],\n        [-0.5469,  0.0096,  0.0908, -0.5938,  0.5469, -0.2871,  0.3027,  0.3027],\n        [-0.1436,  0.5430,  0.4355,  0.2207, -0.0192,  0.2539, -0.5664, -0.1631],\n        [-0.5977, -0.1006, -0.0479, -0.1533, -0.2256,  0.3105,  0.4121,  0.2539],\n        [-0.2578,  0.1338,  0.5625, -0.2207, -0.2539,  0.4414, -0.4941,  0.2637],\n        [ 0.4941, -0.1729,  0.1338,  0.3359, -0.4453, -0.2256,  0.3105,  0.5898]],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_assistant_only - AssertionError: Parameter model.layers.0.self_attn.q_norm.weight has not changed
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f62839d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16,\n       grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1.], device='cuda:0', dtype=torch.bfloat16, requires_grad=True))
   +    where <built-in method allclose of type object at 0x7f62839d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm[trl-internal-testing/tiny-Qwen2VLForConditionalGeneration] - AssertionError: Param model.visual.blocks.0.norm1.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0',\n       dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n        1., 1., 1., 1., 1., 1., 1., 1., 1., 1.], device='cuda:0',\n       dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm[trl-internal-testing/tiny-Qwen2_5_VLForConditionalGeneration] - AssertionError: Param model.visual.blocks.0.norm1.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm[trl-internal-testing/tiny-Qwen3VLForConditionalGeneration] - AssertionError: Param model.visual.blocks.0.norm1.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm[trl-internal-testing/tiny-Gemma3ForConditionalGeneration] - AssertionError: Param model.vision_tower.vision_model.encoder.layers.0.layer_norm1.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7f9f939d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7f9f939d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm_text_only_data[conversational_language_modeling-trl-internal-testing/tiny-Qwen2_5_VLForConditionalGeneration] - AssertionError: Param model.language_model.layers.0.input_layernorm.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm_text_only_data[conversational_prompt_completion-trl-internal-testing/tiny-Qwen2_5_VLForConditionalGeneration] - AssertionError: Param model.language_model.layers.0.input_layernorm.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  FAILED tests/test_sft_trainer.py::TestSFTTrainer::test_train_vlm_text_only_data[standard_prompt_completion-trl-internal-testing/tiny-Qwen2_5_VLForConditionalGeneration] - AssertionError: Param model.language_model.layers.0.input_layernorm.weight is not updated
  assert not True
   +  where True = <built-in method allclose of type object at 0x7fc1a79d6c20>(tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, grad_fn=<CloneBackward0>), Parameter containing:\ntensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       device='cuda:0', dtype=torch.bfloat16, requires_grad=True), rtol=1e-12, atol=1e-12)
   +    where <built-in method allclose of type object at 0x7fc1a79d6c20> = torch.allclose
  = 50 failed, 729 passed, 54 skipped, 31726 warnings, 251 rerun in 1732.88s (0:28:52) =
```
</details>

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -22,38 +22,34 @@ import torch
 @pytest.fixture(autouse=True)
 def set_model_float32_dtype(monkeypatch):
     """Auto-inject float32 dtype for tiny models defined in trl-internal-testing."""
-    from transformers import (
-        AutoModel,
-        AutoModelForCausalLM,
-        AutoModelForImageTextToText,
-        AutoModelForSequenceClassification,
-        AutoProcessor,
-        AutoTokenizer,
-    )
+    from transformers import PreTrainedModel, PreTrainedTokenizerBase, ProcessorMixin
 
-    def create_wrapper(original_method):
-        @wraps(original_method)
-        def wrapper(pretrained_model_name_or_path, *args, **kwargs):
+    def create_classmethod_wrapper(original_classmethod):
+        # Extract the underlying function from the classmethod
+        original_func = original_classmethod.__func__
+
+        @wraps(original_func)
+        def wrapper(cls, pretrained_model_name_or_path, *args, **kwargs):
             # Only inject if model_id is one of trl-internal-testing
-            if "trl-internal-testing" in pretrained_model_name_or_path:
+            if (
+                isinstance(pretrained_model_name_or_path, str)
+                and "trl-internal-testing" in pretrained_model_name_or_path
+            ):
                 if "dtype" not in kwargs:
                     kwargs["dtype"] = "float32"
 
-            return original_method(pretrained_model_name_or_path, *args, **kwargs)
+            return original_func(cls, pretrained_model_name_or_path, *args, **kwargs)
 
-        return wrapper
+        # Re-wrap as classmethod
+        return classmethod(wrapper)
 
-    # Patch all transformers Auto* classes
+    # Patch base classes - this affects all models, tokenizers, and processors
     for cls in [
-        AutoModel,
-        AutoModelForCausalLM,
-        AutoModelForSequenceClassification,
-        AutoTokenizer,
-        AutoProcessor,
-        AutoModelForImageTextToText,
+        PreTrainedModel,
+        PreTrainedTokenizerBase,
+        ProcessorMixin,
     ]:
-        if hasattr(cls, "from_pretrained"):
-            monkeypatch.setattr(cls, "from_pretrained", create_wrapper(cls.from_pretrained))
+        monkeypatch.setattr(cls, "from_pretrained", create_classmethod_wrapper(cls.from_pretrained))
 
 
 @pytest.fixture(autouse=True)

